compiler/src/dotty/tools/dotc/printing/DecompilerPrinter.scala

```diff
@@ -51,4 +51,16 @@ class DecompilerPrinter(_ctx: Context) extends RefinedPrinter(_ctx) {
       }
     }
   }
+
+  override protected def toTextCore[T >: Untyped](tree: Tree[T]): Text = {
+    import untpd.{modsDeco => _, _}
+    tree match {
+      case TypeApply(fun, args) =>
```

Comment:
I think we should have a protected `def` for each type of tree and override the bits we are interested in.,

Reply:
Added it